lib/timers.js

```diff
@@ -305,6 +306,9 @@ Timeout.prototype.unref = function() {
   if (this._handle) {
     this._handle.unref();
   } else if (typeof(this._onTimeout) === 'function') {
+    // Prevent running callback multiple times
+    // when unref() is called during the callback
+    if (this._called) return;
```

Comment:
I think it's probably best to just do this before `if (this._handle) {`, but I'm not 100% sure it matters.